### PR TITLE
add workaround for klog/coredns crash issue

### LINKED ISSUE #85205: 1.15 CoreDNS immediately crash if Kubernetes apiserver is unavailable
**What happened**:

When the apiserver is unavailable, CoreDNS pods crash while trying to write log files to the read only file system.

**What you expected to happen**:

CoreDNS should continue to run even if the apiserver is unavailable for a short period of time. Log files should be written to an emptyDir.

**How to reproduce it (as minimally and precisely as possible)**:

Restart the apiserver. Notice when the apiserver is back up that CoreDNS pod status is either Error, CrashLoopBackOff or Running but Ready is 0/1. Restart count has increased by at least 1.

**Anything else we need to know?**:

This is a regression from #75414. The initial fix was to upgrade CoreDNS from 1.3.1 to 1.5.0, but it was later reverted, as you can see from the history [here](https://github.com/kubernetes/kubernetes/commits/release-1.15/cmd/kubeadm/app/phases/addons/dns/manifests.go). The initial issue then reappeared. A fix was made for 1.14 in pull request #82128 to mount an emptyDir to /tmp to keep the pods from crashing when writing log files. The same fix can be applied in 1.15.

This issue is blocking rollout of 1.15 for us. It would be really nice if the suggested fix could be included in 1.15.6.

**Environment**: Test
- Kubernetes version (use `kubectl version`): v1.15.5
- Cloud provider or hardware configuration: On premise, vmware nodes
- OS (e.g: `cat /etc/os-release`): CentOS Linux release 7.6.1810 (Core)
- Kernel (e.g. `uname -a`): 3.10.0-957.5.1.el7.x86_64
- Install tools: kubeadm
- Network plugin and version (if this is a network-related bug): Canal v3.7.2
- Others:

### PR DESCRIPTION
**Pull request by chrisohaver**
**What type of PR is this?**

 /kind bug

**What this PR does / why we need it**:

Adding a /tmp volume mount resolves the issue in CoreDNS 1.3.1 where CoreDNS will crash when the Kubernetes API shuts down.

This is the same fix as #82128, which was merged into the 1.14 branch.  CoreDNS 1.3.1 is installed by kubeadm/kubeup in K8s 1.14 and 1.15, so the fix should be in 1.15 too.

Fixes #85205

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/addons/dns/coredns/coredns.yaml.in
Comment (rajansandeep) [inline cluster/addons/dns/coredns/coredns.yaml.in]: Extra indent needs to be fixed.
```
- name: tmp
  mountPath: /tmp
```

File: cluster/addons/dns/coredns/coredns.yaml.in
Comment (rajansandeep) [inline cluster/addons/dns/coredns/coredns.yaml.in]: Same here as well, extra indent
```
- name: tmp
  emptyDir: {}
```

Comment (chrisohaver) [reply cluster/addons/dns/coredns/coredns.yaml.in]: Thanks - same copy paste error in the other files too.
And there was also a tab that snuck into manifests.go.

**Comment by rajansandeep** [comment]
/lgtm
Thank you @chrisohaver 
/assign @neolit123 @MrHohn 
cc @kubernetes/patch-release-team

**Comment by neolit123** [comment]
/approve
for kubeadm

**Comment by neolit123** [comment]
thanks @chrisohaver 
ping @kubernetes/patch-release-team

**Review by MrHohn — APPROVED** [review]
/approve

**Comment by drpaneas** [comment]
Thanks for the PR. This fixes the issue indeed.

**Comment by blurpy** [comment]
Is there anything blocking merge of this? Really hoping to see this fix in the next patch release.

**Comment by dims** [comment]
/assign @tpepper 

needs sign off from the patch release team

**Review by dougm — COMMENTED** [review]
The change lgtm, was it not possible to use the automated cherry pick (cherry_pick_pull.sh) for this one?

**Comment by tpepper** [comment]
> The change lgtm, was it not possible to use the automated cherry pick (cherry_pick_pull.sh) for this one?

See also https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md

It's not necessary to re-do this via the cherrypick automation, but just want to FYI in the event in the future you have a choice between using the automation versus manual.

**Comment by dougm** [comment]
@tpepper yes, the manual pick looks correct now, agreed no need for re-do.  In general though, automated cherry-pick should be preferred when there are no merge conflicts.  Helps avoid the manual errors that happened here and git tooling plays better when the original commit hash is maintained.

**Pull request merged**
